hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go

```diff
@@ -8,140 +8,145 @@ import (
+			Namespace: controlPlaneNamespace,
+		},
+	}
 
-	cluster := &capiazure.AzureCluster{
+	azureClusterIdentity := &capiazure.AzureClusterIdentity{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      hcluster.Spec.InfraID,
+			Name:      hcluster.Name,
 			Namespace: controlPlaneNamespace,
 		},
 	}
-	if _, err := createOrUpdate(ctx, client, cluster, func() error {
-		if cluster.Annotations == nil {
-			cluster.Annotations = map[string]string{}
-		}
-		cluster.Annotations[capiv1.ManagedByAnnotation] = "external"
 
-		cluster.Spec.Location = hcluster.Spec.Platform.Azure.Location
-		cluster.Spec.ResourceGroup = hcluster.Spec.Platform.Azure.ResourceGroupName
-		cluster.Spec.NetworkSpec.Vnet.ID = hcluster.Spec.Platform.Azure.VnetID
-		cluster.Spec.NetworkSpec.Vnet.Name = hcluster.Spec.Platform.Azure.VnetName
-		cluster.Spec.NetworkSpec.Vnet.ResourceGroup = hcluster.Spec.Platform.Azure.ResourceGroupName
-		cluster.Spec.SubscriptionID = hcluster.Spec.Platform.Azure.SubscriptionID
+	if _, err := createOrUpdate(ctx, client, azureClusterIdentity, func() error {
+		return reconcileAzureClusterIdentity(ctx, client, hcluster, azureClusterIdentity, controlPlaneNamespace)
+	}); err != nil {
+		return nil, fmt.Errorf("failed to reconcile Azure cluster identity: %w", err)
+	}
 
-		cluster.Status.Ready = true
-		return nil
+	if _, err := createOrUpdate(ctx, client, azureCluster, func() error {
+		return reconcileAzureCluster(azureCluster, hcluster, apiEndpoint, controlPlaneNamespace)
 	}); err != nil {
-		return nil, fmt.Errorf("failed to upsert Azure capi cluster: %w", err)
+		return nil, fmt.Errorf("failed to reconcile Azure CAPI cluster: %w", err)
 	}
 
-	return cluster, nil
+	return azureCluster, nil
 }
 
-func (a *Azure) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, hcp *hyperv1.HostedControlPlane) (*appsv1.DeploymentSpec, error) {
-	image := providerImage
+func (a Azure) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, _ *hyperv1.HostedControlPlane) (*appsv1.DeploymentSpec, error) {
+	image := a.capiProviderImage
 	if envImage := os.Getenv(images.AzureCAPIProviderEnvVar); len(envImage) > 0 {
 		image = envImage
 	}
 	if override, ok := hcluster.Annotations[hyperv1.ClusterAPIAzureProviderImage]; ok {
 		image = override
 	}
-	return &appsv1.DeploymentSpec{Template: corev1.PodTemplateSpec{Spec: corev1.PodSpec{
-		Containers: []corev1.Container{{
-			Name:    "manager",
-			Image:   image,
-			Command: []string{"/manager"},
-			Args: []string{
-				"--namespace=$(MY_NAMESPACE)",
-				"--leader-elect=true",
-			},
-			Resources: corev1.ResourceRequirements{
-				Requests: corev1.ResourceList{
-					corev1.ResourceCPU:    resource.MustParse("10m"),
-					corev1.ResourceMemory: resource.MustParse("10Mi"),
+	defaultMode := int32(0640)
+	return &appsv1.DeploymentSpec{
+		Template: corev1.PodTemplateSpec{
+			Spec: corev1.PodSpec{
+				TerminationGracePeriodSeconds: k8sutilspointer.Int64(10),
+				Tolerations: []corev1.Toleration{
+					{
```

Comment:
I assume this is carried over from somewhere else? we do not need this toleration.

Reply:
Yeah it was leftover. I will remove it.